### ISSUE TITLE
Extend mlx with Additional Optimizers

### ISSUE THREAD

**Issue by m0saan** [ml-explore/mlx#46]
**Issue Type:** Enhancement

**Proposal:**
I would like to propose an enhancement to the framework by adding a variety of optimizers. While the current implementation includes Stochastic Gradient Descent (SGD) and Adam optimizers, the addition of more optimization algorithms could greatly benefit users with diverse requirements.

**Potential Optimizers to Include:**
1. **Adagrad:** 
2. **RMSprop:** 
3. **Adadelta:** 
4. **Nadam:** 
5. **AdamW**
etc ...

I would appreciate your guidance on the next steps for contributing to this enhancement. Whether it's providing additional details, discussing the proposed optimizers, or collaborating on the implementation, I am eager to contribute to the growth of this fantastic framework.

**Comment by awni** [comment]
Thanks, we'd be happy to take PRs for any of the above optimizers. I would suggest starting with a draft PR for one of the above, maybe AdamW and we can go from there?

Generally we follow the PyTorch API for `mlx.optim`. If there are any involved design questions then we can use this thread to discuss (or even better a draft PR) to discuss.

**Comment by m0saan** [comment]
I'm currently working on a PR to integrate the AdamW optimizer (and other next!). Once ready, I'll submit it for reviews. Your feedback is highly appreciated! 🚀 Thanks for your support! :p

**Comment by awni** [comment]
Hi - thanks a ton! But FYI there is a PR for AdamW already https://github.com/ml-explore/mlx/pull/72 !

Maybe try Adagrad / RMSProp / or adding Nesterov style momentum to SGD instead?

**Comment by m0saan** [comment]
On it!
